### ISSUE TITLE
hybrid strategy isn't working

### ISSUE THREAD

**Issue by fluxoz** [fluxoz/ScalpEdge#33]
23:40:21 [INFO] scalpedge.main — ScalpEdge backtester starting ...
23:40:21 [INFO] scalpedge.main — Tickers: SPY, TSLA
23:40:21 [INFO] scalpedge.main — ML layer disabled (torch/scikit-learn not installed). Install with: uv sync --extra ml  or  pip install -e '.[ml]'
23:40:22 [INFO] scalpedge.data — SPY: data is up-to-date (3276 bars total)
23:40:22 [INFO] scalpedge.main — SPY benchmark loaded: 3276 bars (for regime filter).
23:40:22 [INFO] scalpedge.main — ============================================================
23:40:22 [INFO] scalpedge.main —   Processing ticker: SPY
23:40:22 [INFO] scalpedge.main — ============================================================
23:40:22 [INFO] scalpedge.main — [SPY] Loading 5-minute OHLCV data ...
23:40:22 [INFO] scalpedge.data — SPY: data is up-to-date (3276 bars total)
23:40:22 [INFO] scalpedge.main — [SPY] 3276 bars loaded.
23:40:22 [INFO] scalpedge.main — [SPY] Computing TA indicators ...
23:40:22 [INFO] scalpedge.main — [SPY] 3262 bars after indicator warm-up.
23:40:22 [INFO] scalpedge.main — [SPY] Running TA-only baseline ...
=======================================================
  SPY  |  Strategy: ta_only
=======================================================
  Trades          : 216
  Win Rate        : 35.2%
  Avg Win         : 0.139%
  Avg Loss        : -0.131%
  Expectancy      : -0.036%
  Profit Factor   : 0.58
  Sharpe Ratio    : -6.97
  Max Drawdown    : -8.08%
  Total Return    : -7.50%
  CAGR            : -37.48%
=======================================================
23:40:22 [INFO] scalpedge.main — [SPY] Fitting ML models on 2609 bars (train set) ...
23:40:22 [INFO] scalpedge.main — [SPY] Running Hybrid backtest on 653 bars (test set) ...
23:40:22 [WARNING] scalpedge.backtester — No trades generated for SPY / hybrid
=======================================================
  SPY  |  Strategy: hybrid
=======================================================
  Trades          : 0
  Win Rate        : 0.0%
  Avg Win         : 0.000%
  Avg Loss        : 0.000%
  Expectancy      : 0.000%
  Profit Factor   : 0.00
  Sharpe Ratio    : 0.00
  Max Drawdown    : 0.00%
  Total Return    : 0.00%
  CAGR            : 0.00%
=======================================================
23:40:22 [INFO] scalpedge.main — [SPY] Equity curve saved → data/SPY_equity_curve.png
23:40:22 [INFO] scalpedge.main — ============================================================
23:40:22 [INFO] scalpedge.main —   Processing ticker: TSLA
23:40:22 [INFO] scalpedge.main — ============================================================
23:40:22 [INFO] scalpedge.main — [TSLA] Loading 5-minute OHLCV data ...
23:40:22 [INFO] scalpedge.data — TSLA: data is up-to-date (3276 bars total)
23:40:22 [INFO] scalpedge.main — [TSLA] 3276 bars loaded.
23:40:22 [INFO] scalpedge.main — [TSLA] Computing TA indicators ...
23:40:22 [INFO] scalpedge.main — [TSLA] 3262 bars after indicator warm-up.
23:40:22 [INFO] scalpedge.main — [TSLA] Running TA-only baseline ...
=======================================================
  TSLA  |  Strategy: ta_only
=======================================================
  Trades          : 222
  Win Rate        : 40.1%
  Avg Win         : 0.309%
  Avg Loss        : -0.342%
  Expectancy      : -0.081%
  Profit Factor   : 0.60
  Sharpe Ratio    : -5.77
  Max Drawdown    : -16.94%
  Total Return    : -16.73%
  CAGR            : -66.82%
=======================================================
23:40:22 [INFO] scalpedge.main — [TSLA] Market regime filter ENABLED — using SPY 5-bar rolling VWAP.
23:40:22 [INFO] scalpedge.main — [TSLA] Fitting ML models on 2609 bars (train set) ...
23:40:22 [INFO] scalpedge.main — [TSLA] Running Hybrid backtest on 653 bars (test set) ...
23:40:23 [WARNING] scalpedge.backtester — No trades generated for TSLA / hybrid
=======================================================
  TSLA  |  Strategy: hybrid
=======================================================
  Trades          : 0
  Win Rate        : 0.0%
  Avg Win         : 0.000%
  Avg Loss        : 0.000%
  Expectancy      : 0.000%
  Profit Factor   : 0.00
  Sharpe Ratio    : 0.00
  Max Drawdown    : 0.00%
  Total Return    : 0.00%
  CAGR            : 0.00%
=======================================================
23:40:23 [INFO] scalpedge.main — [TSLA] Equity curve saved → data/TSLA_equity_curve.png
23:40:23 [INFO] scalpedge.main —

All done.
23:40:23 [INFO] scalpedge.main — Summary:
23:40:23 [INFO] scalpedge.main —   SPY: OK
23:40:23 [INFO] scalpedge.main —   TSLA: OK

I have ran uv sync --extra ml and it's still not working
